### ISSUE TITLE
bug: message stuck in sending status for a long time

### ISSUE THREAD

**Issue by chaitanyaprem** [status-im/status-go#5449]
# Problem

Have noticed this multiple times and even during dogfooding session, a message that is sent whose status is still shown as sending. 
Eventhough the message is received by other side in 1:1 chat, group chat and in this case community. 

Attaching the logs for reference.
MessageID of the message is `0xf37863ffcb2fcc6d7d05653a25db2ef4d808fa4e1a6cf584b5ddc4587ff82648`

Even after 5 minutes, message is still shown as sending status.

# Acceptance Criteria

Message should not be showing sending status for soo long especially if it is already delivered. 

# Notes

![image](https://github.com/status-im/status-go/assets/20137744/59569744-c393-4c54-af31-07aa42983655)

[geth.log.gz](https://github.com/user-attachments/files/16048294/geth.log.gz)

**Comment by chaitanyaprem** [comment]
Uploading more logs @kaichaosun 
[geth.log.gz](https://github.com/user-attachments/files/16048577/geth.log.gz)

**Comment by kaichaosun** [comment]
The timeline for message `0xf37863ffcb2fcc6d7d05653a25db2ef4d808fa4e1a6cf584b5ddc4587ff82648`,
- sent out at ~11:04~ 11:01:22
- store hash query at ~11:07~ 11:04:33
- the success of query result at ~11:11~ 11:07:19
- EventEnvelopeSent triggered at ~11:11~ 11:07:19

The period of ~3s~ **3minutes** between `send out` and `send query request` is ~expected~ unexpected, and it should take up to 6 seconds.
The period of ~4s~ **3minutes** between `send query request` and `get request response` is also too long and unexpected. 

~But it only takes around 7 seconds~, do you mean it's still showing `sending` after 5 minutes for this message or is there other message for this long delay? @chaitanyaprem

**Comment by chaitanyaprem** [comment]
> But it only takes around 7 seconds, do you mean it's still showing `sending` after 5 minutes for this message or is there other message for this long delay? @chaitanyaprem

It was for this message, screenshot was taken after 5 minutes of sending the message around local time 11:06AM and it was still showing as sending. I did not notice when status has finally changed, but now (after an hour and half) i don't see sending any more.

**Comment by kaichaosun** [comment]
It seems the store request takes really long, also the relay may also takes long since it's adding the message hash for query only after relay publish.

The request ID for hash query is `0x65d92cd709449fabc5f9681e38906f213556bf2572f673431ff1fb51a0a9b1be`, @Ivansete-status you may know how to get the timeline of the query in nwaku server? 

@chaitanyaprem It looks like the network connection can be the cause of the issue, or maybe I'm looking into the wrong branch, where can I find the version you are using for this test?

**Comment by kaichaosun** [comment]
I find the some info in elastic, [here](https://kibana.infra.status.im/app/discover#/?_a=(columns:!(message,host,syslogtag,logsource),filters:!(),index:'57c3ac60-8ac3-11ec-b597-d13e4349f051',interval:auto,query:(language:kuery,query:'65d92cd709449fabc5f9681e38906f213556bf2572f673431ff1fb51a0a9b1be'),sort:!(!('@timestamp',desc)))&_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now%2Fd,to:now%2Fd)))

<img width="1303" alt="image" src="https://github.com/status-im/status-go/assets/10568673/aa9458ce-6020-4695-9002-1e3f097387e4">

It takes pretty long to get the response.

**Comment by chaitanyaprem** [comment]
> I find the some info in elastic, [here](https://kibana.infra.status.im/app/discover#/?_a=(columns:!(message,host,syslogtag,logsource),filters:!(),index:'57c3ac60-8ac3-11ec-b597-d13e4349f051',interval:auto,query:(language:kuery,query:'65d92cd709449fabc5f9681e38906f213556bf2572f673431ff1fb51a0a9b1be'),sort:!(!('@timestamp',desc)))&_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now%2Fd,to:now%2Fd)))
> <img alt="image" width="1303" src="https://private-user-images.githubusercontent.com/10568673/344590818-aa9458ce-6020-4695-9002-1e3f097387e4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk4OTU1ODQsIm5iZiI6MTcxOTg5NTI4NCwicGF0aCI6Ii8xMDU2ODY3My8zNDQ1OTA4MTgtYWE5NDU4Y2UtNjAyMC00Njk1LTkwMDItMWUzZjA5NzM4N2U0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAyVDA0NDEyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIxMmM2NDkxOGE2ZTJiNWJlMzBkN2RmMDNhNjY5OGFjYTkwNDc1YmY0NDI2NzBjMjIwYWI2NDFmNGMyYWZhYjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kSUIzsGFsWZ1HokuJk5B971F_iKBwXrStXs7_XclHQw">
> 
> It takes pretty long to get the response.

Wow....that is a long time. cc @Ivansete-status store query took close to 3 mins.

**Comment by chaitanyaprem** [comment]
I am also wondering what is the behaviour for the following scenario? The reason for this query is we have observed this behaviour even in case of 1:1 chat during our dogfooding session.

What if userA sends a message to UserB in 1:1 chat and mvds also acks receipt of message by UserB. 
Will we still depend on store-query to change sending status to sent at UserA's side? If so, then we need to fix it. 

Ideally status can be changed by any sort of ack either by mvds, store-query or e2e protocol that is being built at a later point in time.

**Comment by kaichaosun** [comment]
> Will we still depend on store-query to change sending status to sent at UserA's side? If so, then we need to fix it.

No, the ack message will also change the state from `outging` (or `sending`) to `sent`.

> Ideally status can be changed by any sort of ack either by mvds, store-query or e2e protocol that is being built at a later point in time.

This is a good idea, we need to encapsulate the interface for better usage in different reliability solutions.

**Comment by fryorcraken** [comment]
> > Ideally status can be changed by any sort of ack either by mvds, store-query or e2e protocol that is being built at a later point in time.
> 
> This is a good idea, we need to encapsulate the interface for better usage in different reliability solutions.

Are we tracking this somewhere?

**Comment by kaichaosun** [comment]
@fryorcraken not yet, it could be one of the requirements when implementing e2e reliability. I believe it may include a lot of refactors within status-go, for example mvds. Or we can make e2e reliability and mvds both in waku-sdk.

**Comment by Ivansete-status** [comment]
As it can be seen in the description of the following issue, the `messageHash` queries are taking more than 5 minutes, on average (see the second query:) https://github.com/waku-org/nwaku/issues/2895

@kaichaosun , @chaitanyaprem - ~~is this happening only with store-v3?~~ edited: I confirmed that only happens from `store-v3`

**Comment by kaichaosun** [comment]
This is fixed with the improvements in nwaku.

**Comment by kaichaosun** [comment]
Hey @Ivansete-status I noticed there are [store request](https://kibana.infra.status.im/app/discover#/?_a=(columns:!(message,host,syslogtag,logsource),filters:!(),index:'57c3ac60-8ac3-11ec-b597-d13e4349f051',interval:auto,query:(language:kuery,query:'4424c86380d1a43ea3424c7dccb3d6c6f3d999c8670c48ccdff9c3de0272da0b'),sort:!(!('@timestamp',desc)))&_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now%2Fd,to:now%2Fd))) takes more than 30s, is it expected?

**Comment by Ivansete-status** [comment]
Hi @kaichaosun ! Yes, we have right now queries that take some minutes. That will be fixed when https://github.com/waku-org/nwaku/issues/2895 is deployed ( this month , August )

**Comment by Ivansete-status** [comment]
@chaitanyaprem - a new version with a fix candidate has been deployed in `status.prod`. Kindly check whether the issue is sorted out. Thanks!
